<launch>
    <arg name="rviz" default="true" />

    <!-- start the Realsense -->
    <include file="$(find realsense2_viewer)/launch/realsense2_viewer.launch" >
        <arg name="json_file_path" value="$(find realsense2_viewer)/config/D435i-pick.json"/>
        <arg name="width" value="640" />
        <arg name="height" value="480" />
        <arg name="clip_distance" value="0.8" />
    </include>

    <!-- start the rviz -->
    <node if="$(arg rviz)" name="$(anon rviz)" pkg="rviz" type="rviz" respawn="false" args="-d $(find gpd_ros)/launch/gpd_pointnet_realsense_service.rviz" output="screen" />

    <!-- detect grasps -->
	<node name="detect_grasps_server" pkg="gpd_ros" type="detect_grasps_server_pointnet_realsense" output="screen">
 
       <!-- What type of point cloud is used and what ROS topic it comes from -->
       <param name="cloud_topic" value="/camera/depth/color/points" />

       <!-- Filepath to the configuration file for GPD -->
       <param name="config_file" value="$(find gpd_ros)/cfg/python_classifier_params.cfg" />

       <!-- Rviz plotting parameters -->
       <param name="rviz_topic" value="grasps_rviz" />
	</node>

</launch>
